ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,462 @@
+
+        """
+        maxsize = self.settings.get('max_mosaic_size', 1e8)  # Default 10k x 10k
+        image = self.chinfo.datasrc[key]
+
+        # Prevent writing very large mosaic
+        if (image.width * image.height) > maxsize:
+            s = 'Mosaic too large to be written {0}'.format(image.shape)
+            self.w.status.set_text(s)
+            self.logger.error(s)
+            return
+
+        # Insert mosaic data and header into output HDU
+        hdu = fits.PrimaryHDU(image.get_data())
+        self._write_header(image, hdu)
+
+        # Write history to PRIMARY
+        self._write_history(key, hdu)
+
+        # Write to file
+        hdu.writeto(outfile, clobber=True)
+
+    def _write_mef(self, key, extlist, outfile):
+        """Write out regular multi-extension FITS data."""
+        with fits.open(outfile, mode='update') as pf:
+            # Process each modified data extension
+            for idx in extlist:
+                k = '{0}[{1}]'.format(key, self._format_extname(idx))
+                image = self.chinfo.datasrc[k]
+
+                # Insert data and header into output HDU
+                pf[idx].data = image.get_data()
+                self._write_header(image, pf[idx])
+
+            # Write history to PRIMARY
+            self._write_history(key, pf['PRIMARY'])
+
+    def toggle_save_cb(self, w, res_dict):
+        """Only enable saving if something is selected."""
+        if len(res_dict) > 0:
+            self.w.save.set_enabled(True)
+        else:
+            self.w.save.set_enabled(False)
+        return
+
+    def save_images(self):
+        """Save selected images.
+
+        This uses Astropy FITS package to save the outputs no matter
+        what user chose to load the images.
+
+        """
+        res_dict = self.treeview.get_selected()
+        clobber = self.settings.get('clobber', False)
+        self.treeview.clear_selection()  # Automatically disables Save button
+
+        # If user gives empty string, no suffix.
+        if self.suffix:
+            sfx = '_' + self.suffix
+        else:
+            sfx = ''
+
+        # Process each selected file. Each can have multiple edited extensions.
+        for infile in res_dict:
+            f_pfx = os.path.splitext(infile)[0]  # prefix
+            f_ext = '.fits'  # Only FITS supported
+            oname = f_pfx + sfx + f_ext
+            outfile = os.path.join(self.outdir, oname)
+
+            s = 'Writing out {0} to {1} ...'.format(infile, oname)
+            self.w.status.set_text(s)
+            self.logger.debug(s)
+
+            if os.path.exists(outfile) and not clobber:
+                self.logger.error('{0} already exists'.format(outfile))
+                continue
+
+            bnch = res_dict[infile]
+
+            if bnch.path is None or not os.path.isfile(bnch.path):
+                self._write_mosaic(f_pfx, outfile)
+            else:
+                shutil.copyfile(bnch.path, outfile)
```

Comment:
It overwrites the existing file. But by the time it gets to this line, it is already very clear that user wants to clobber the old file, because there is a check for clobber setting on Line 401 above.